### ISSUE TITLE
Upload to Blob: Http-Response 403 if last upload was not completed

### ISSUE THREAD

**Issue by raycon94** [Azure/azure-iot-sdk-c#761]
- **OS and version used:** keil rtx v4 

- **SDK version used:** 1.2.10 

# Description of the issue:

Hello,
when the last uploading to the blob Fails, for Example when the Connection is down. I am not able to resend the data because I get the Status = 403.
Is it because the last data-stream was not completed properly? If yes, how can I complete the last stream?

**Comment by jubueche** [comment]
I am having the same problem. The first time, the upload is complete (after that core panic'ed for some reason idk yet) and after that I get a 403.

**Comment by jubueche** [comment]
Update: So it seems like when I wairt for an hour more or less, it works again, meaning uploading, but then crashing.
Unfortunatley my stack trace decoder doesn't give me an exception for this:

0x400d6171: _svfprintf_r at /Users/ivan/e/newlib_xtensa-2.2.0-bin/newlib_xtensa-2.2.0/xtensa-esp32-elf/newlib/libc/stdio/../../../.././newlib/libc/stdio/vfprintf.c:1529

0x4015eb2e: _vsnprintf_r at /Users/ivan/e/newlib_xtensa-2.2.0-bin/newlib_xtensa-2.2.0/xtensa-esp32-elf/newlib/libc/stdio/../../../.././newlib/libc/stdio/vsnprintf.c:72

0x4015eb6a: vsnprintf at /Users/ivan/e/newlib_xtensa-2.2.0-bin/newlib_xtensa-2.2.0/xtensa-esp32-elf/newlib/libc/stdio/../../../.././newlib/libc/stdio/vsnprintf.c:41

0x400f3ff2: STRING_construct_sprintf at /home/julian/eclipse-workspace/chaze-esp32/components/esp-azure/azure-iot-sdk-c/c-utility/src/strings.c:786

0x400ecac0: IoTHubClient_LL_UploadMultipleBlocksToBlob_Impl at /home/julian/eclipse-workspace/chaze-esp32/components/esp-azure/azure-iot-sdk-c/iothub_client/src/iothub_client_ll_uploadtoblob.c:910

0x400ebb8d: IoTHubClientCore_LL_UploadMultipleBlocksToBlobEx at /home/julian/eclipse-workspace/chaze-esp32/components/esp-azure/azure-iot-sdk-c/iothub_client/src/iothub_client_core_ll.c:2595

0x400ea576: IoTHubDeviceClient_LL_UploadMultipleBlocksToBlob at /home/julian/eclipse-workspace/chaze-esp32/components/esp-azure/azure-iot-sdk-c/iothub_client/src/iothub_device_client_ll.c:127

0x40090c1a: upload_to_blob_block at /home/julian/eclipse-workspace/chaze-esp32/components/upload_to_blob_block/upload_to_blob_block.c:118

0x400d38ba: azure_task at /home/julian/eclipse-workspace/chaze-esp32/main/main.c:89

0x4008de7d: vPortTaskWrapper at /home/julian/Documents/esp/esp-idf/components/freertos/port.c:403

**Comment by jubueche** [comment]
Another thing that might help: I am using this git-hub repo in order to use the ESP with azrue: https://github.com/espressif/esp-azure

**Comment by sergaz-msft** [comment]
@jubueche I have answered your question on Stack Overflow: https://stackoverflow.com/questions/54245181/azure-iot-sdk-c-error-401-when-connecting-to-blob-storage/54345333#54345333

**Comment by jspaith** [comment]
@raycon94 / @jubueche - has @sergaz-msft's SO answer address your issue here?  Thanks

**Comment by jubueche** [comment]
No, but the issue was resolved. It was a Null Pointer exception that wasn’t caught by your automatic testing. I fixed it and made a pull request some time ago that is still pending.

**Comment by ewertons** [comment]
@raycon94 ,
you have a maximum number of outstanding uploads that is maintained within your IoT Hub.
You can control that number. In your iothub, go to "File Update" and control it with "Maximum delivery count".

Downloads that have been interrupted by some reason (network drop, for example) are not notified as complete to the IoT Hub, getting stale. They have a timeout range in the IoT Hub though, which by default is 1 hour (that's why someone else mentioned above that after 1 hour they were able to do upload to blob again).

So to recap, if you have 10 uploads failing in a row because of some network issue, the 11th upload and on will fail until each of the 10 previous uploads gets timed out in the IoT Hub control list.

There is no option at this time to circumvent this limit other than the settings in your IoT Hub (as mentioned above).

Let us know if we can assist in any other way.

**Comment by ewertons** [comment]
Hi @raycon94 ,
we will close this issue for now, but please feel free to reopen it if you would like to follow up.
Thanks,
Azure IoT SDK Team

**Comment by az-iot-builder-01** [comment]
@raycon94, @jubueche, @jspaith, @ewertons, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)

**Comment by raycon94** [comment]
Hello @ewertons,

sorry for my delayed Response, I was busy with other projects. 
Is it possible to send an packet to the Cloud which indicates that the last upload-process has finished?

Thanks,
Raycon94
